### ISSUE TITLE
Extract config/base plugin valid for any language/type from Golang

### ISSUE THREAD

**Issue by camilamacedo86** [kubernetes-sigs/kubebuilder#2015]
**What do you want to happen?**

A plugin base which would be responsible to scaffold what is common for any kind of plugin no matter the language and what be integrated with KB and use it's as a consumer. ( basically, it is the hack and config directory since the PROJECT file is scaffold by the CLI ) 

It would allow the community to create any language/type plugin using the same base/config defined by KB in order to ensure its compatibility and maintainability see https://github.com/operator-framework/operator-sdk/issues/2745.Following the requirements/problems that also could be addressed with:

- Make SDK use the real plugin system instead of having to do “hacky” wrappers around the `go.kubebuilder.io` plugin. To SDK Helm/Ansible plugins it would actually mean only implement what which is specifically for its types which means that SDK project would no longer need to copy and paste the config directory boilerplates from kubebuilder in order to do the same. 
- It could allow we init hybrid operators such as; `layout: config/v1,helm/v1,go/v3` which would have the init scaffold of all plugins. See: https://github.com/operator-framework/operator-sdk/issues/670
- Solve the concerns issues raised in the https://github.com/operator-framework/enhancements/pull/44 regarding the risks of the language plugins deviated too much from the upstream base and to keep they maintainable. 

**Is there currently another issue associated with this?**

This issue would be a follow up of #1991 and  #1941

**backwords compatibility**

In order to ensure the backwords compatibility if we call the init command by informing only a plugin that is not the base config, the CLI needs to scaffold the base + the config as it is done currently. E.g `operator-sdk init --plugins=helm/v2` will scaffold the base+helm

**Comment by camilamacedo86** [comment]
Hi @Adirio, 

I tracked the issue regarding the scenario that we spoke about. 

wdyt @estroz @DirectXMan12

**Comment by camilamacedo86** [comment]
The idea proposed regarding the project has a config-gen that also might able to address these requirements. See: https://github.com/kubernetes-sigs/kubebuilder/pull/1831. We probably also might able to replace the base/config with the config-gen when it is stable. c/c @pwittrock

**Comment by camilamacedo86** [comment]
As we discussed today before we decide to move forward with this or not we need first have a better position over #1831:

- shows that @pwittrock like the idea of adding a sample ( project-v3-config-gen in the testdata which would be == project-v3 but using config-gen for we have better clarity of how it will work, what is covered or not already by it and etc)
- my (@camilamacedo86) concerns is regarding the UX to apply customizations on the file. However, maybe config-gen also can write them in the disk instead of only allow patches.

So, it is blocked by we check out the config-gen final stage in KB to address all concerns about it first.

**Comment by camilamacedo86** [comment]
After the controller-tool meeting, this topic was discussed with @pwittrock, @DirectXMan12 and @Adirio where I understand that we reach a consensus that: 

- #1831 - config-gen has good potential to address these needs
- The @pwittrock idea is we add the config-gen as alpha to allows it starts to be tested, used and improved until it reaches the wished maturity 
- we have not a possible ETA for config-gen be stabilized and covering all options/scenarios for we are able to see if we could find a way to replace the config/ directory by it and keep the backwards capabilities with the current plugins
- it still a possible good strategy for indeed we decouple the kustomize config from the plugin and we try to replace it by config-gen in the future. 

We need to define if we will release 3.0.0 before or after the changes: https://github.com/kubernetes-sigs/kubebuilder/issues/2043. However, after these changes, this option shows be able to be achieved with a low effort. 

So, imo it still the best approach discussed so far to allow us to move forward, make it easier for the other plugins keep maintained and others are able to create plugins with are compatible with SDK/KB for any language/stack that might please them and indeed to try to replace that by config-gen in the future.

**Comment by camilamacedo86** [comment]
POC done in the past to check if we could centralize the config. See: https://github.com/operator-framework/operator-sdk/pull/3815/files

**Comment by camilamacedo86** [comment]
It also solves the scenario: https://github.com/operator-framework/operator-sdk/issues/4542 for example.
